jar.go

```diff
@@ -295,6 +299,101 @@ func (j *Jar) cookies(u *url.URL, now time.Time) (cookies []*http.Cookie) {
 	return cookies
 }
 
+// AllCookies returns all cookies for the given URL, or a subpath of the given URL.
+// The returned cookies will have Domain, Expires, HttpOnly, Name,
+// Secure, Path, and Value filled out.
+//
+// It returns an empty slice if the URL's scheme is not HTTP or HTTPS.
+func (j *Jar) AllCookies(u *url.URL) (cookies []*http.Cookie) {
```

Comment:
Hmm, the distinction between this and Cookies wasn't clear to me on first reading. I wonder if the behaviour of this is a little over-tailored to what you're trying to do now.

How about having a similar method, also called AllCookies but remove the URL argument and really have it return all cookies in the jar (I guess you'll need to avoid returning expired ones)?

Then the semantics match the method name better and we can avoid duplicating a lot of the Cookies logic here.

To help in your specific use case, you could also provide a helper function (it might need to be a method I guess):

    // UnderURL reports whether the given cookie is under
    // the given URL.
    func UnderURL(c *http.Cookie, u *url.URL) bool

but at the moment I'm sure whether that's not trivial to do outside
of this package.